### PR TITLE
Update yap.rb

### PR DESCRIPTION
**Pull request by Zialus**
Added patch to make _stable_ version compile again.
Updated _devel_ version.
Added _HEAD_ version.

Dependencies and brew options still need improvements, but this should be a good start to get yap out of the boneyard.

### CODE REVIEW

File: yap.rb [hunk 1]
```diff
@@ -1,39 +1,77 @@
...
+
+    patch do
+      url "https://gist.githubusercontent.com/Zialus/6fb48baa014daca6f530db33c0839e67/raw/7c02ec8c0a42baa93195398b198c6ae09395c0b6/yap-6.3.4.patch"
+      sha256 "85fa385a082d5556f5cbb851f2a7a044d9c1c8eeff659a618faa1029cff564ef"
+    end
   end
 
-  bottle :disable, "Fails with various compilers"
+  head do
+    url "https://github.com/vscosta/yap-6.3.git"
```
Comment (fxcoudert) [inline yap.rb]: The project webpage says that the git repo is `git://yap.dcc.fc.up.pt/yap-6`

File: yap.rb [hunk 2]
```diff
@@ -1,39 +1,77 @@
...
+
+    fails_with :clang do
+      cause "uses variable-length arrays in structs, which will never be supported by clang"
+    end
+  end
 
   devel do
-    url "https://www.dcc.fc.up.pt/~vsc/Yap/yap-6.3.3.tar.gz"
-    sha256 "aee3b449b1669af07a8291ce6e7fb0a9b35e1343e2ab96fadb1a37552397fa78"
+    url "https://github.com/vscosta/yap-6.3/archive/YAP-6.3.4.tar.gz"
```
Comment (fxcoudert) [inline yap.rb]: Webpage says latest version is 6.3.3…

Comment (ilovezfs) [inline yap.rb]: also 6.3.4 is designated as pre-release on GitHub

File: yap.rb (hunk 1 above)
Comment (Zialus) [inline yap.rb]: There's a new WIP [webpage](https://www.dcc.fc.up.pt/~vsc/yap).

Notice the lower case "Y".

I'm a student at DCC - FCUP btw. That's why i know about this.

File: yap.rb (hunk 2 above)
Comment (Zialus) [inline yap.rb]: 6.3.3 had the same status.

compare: 

> The latest development version of Yap-6 is yap-6.3.4 and can be obtained from the repositories

source: https://www.dcc.fc.up.pt/~vsc/yap/index.html#download

with:

>  The latest development version of Yap-6 is yap-6.3.3 and can be obtained as:

source: https://www.dcc.fc.up.pt/~vsc/Yap/downloads.html

**Comment by MikeMcQuaid** [comment]
Closing as we've deprecated Homebrew/boneyard but thanks for the PR anyway!

**Pull request closed without merging**